pkg/beacon/analyzer.go

```diff
@@ -323,31 +327,47 @@ func getTsHistogramScore(min int64, max int64, tsList []int64) ([]int64, []int,
 		cv = 1.0
 	}
 
-	// calculate first potential score
-	// histograms with bigger flat sections will score higher, up to 4 flat sections
-	// this will score well for graphs that have flat sections with a big distance between them,
-	// i.e. a beacon that alternates between 1 and 5 connections per hour
-	// This score will only be calculated if the number of total bars on the histogram fills at
-	// least half the day (>11), otherwise this could lead to a false positive and not fill the
+	cvScore := math.Ceil((1.0-float64(cv))*1000) / 1000
+	if cvScore > 1.0 {
+		cvScore = 1.0
+	}
+
+	// Calculate second potential score
+	// this will score well for graphs that have 2-3 flat sections in their connection histogram,
+	// or a bimodal freqCount histogram.
+	// Example - a beacon that alternates between 1 and 5 connections per hour
+	// This score will only be calculated if the number of total bars on the histogram is at
+	// least 10, otherwise this could lead to a false positive and not fill the
 	// original intent of this scoring.
-	score1 := 0.0
+	bimodalFit := float64(0)
+
+	if totalBars > 10 {
+		largest := 0
+		secondLargest := 0
+
+		// get top two frequency mode bars
+		for key, value := range freqCount {
+			if key > 0 {
+				if value > largest {
+					secondLargest = largest
+					largest = value
+				} else if value > secondLargest {
+					secondLargest = value
+				}
 
-	if totalBars > 11 {
-		score1 = math.Ceil((float64(4)/float64(len(freqCount)))*1000) / 1000
-		if score1 > 1.0 {
-			score1 = 1.0
+			}
 		}
+
+		// calculate the percentage of hour blocks that fit into the top two mode buckets
+		bimodalFit = float64(largest+secondLargest) / float64(totalBars-1)
```

Comment:
The thought here was to provide a little buffer for the score by throwing out one potential outlier bucket. The common outlier I had in mind was a smaller connection count at the start or end of a beacon session. 

For example, if the connections started 10 minutes before the 4th hour bucket, the graph might look like:
__.|||||||||||||||||

This allows that third hour (the . in my drawing above) to be excluded.

I could make that value a constant to make it more readable -- something like REMOVE_OUTLIERS = 1 

This was also shown in the graph in case 4 of my examples (Uniform Distribution with Outlier).

Reply:
That makes sense. Please make that value a constant :+1: